File: tbsrc-golang/pkg/types/order_stats.go
package types

// OrderStats 对应 C++ struct OrderStats
// 参考: tbsrc/Strategies/include/ExecutionStrategyStructs.h:44-68
type OrderStats struct {
	Active     bool   // m_active
	New        bool   // m_new
	ModifyWait bool   // m_modifywait
	Cancel     bool   // m_cancel
	Modify     int32  // m_modify
	LastTS     uint64 // m_lastTS
	OrderID    uint32 // m_orderID
	OldQty     int32  // m_oldQty
	NewQty     int32  // m_newQty
	Qty        int32  // m_Qty
	OpenQty    int32  // m_openQty
	CxlQty    int32  // m_cxlQty
	DoneQty    int32  // m_doneQty
	QuantAhead float64 // m_quantAhead
	QuantBehind float64 // m_quantBehind
	Price      float64 // m_price
	NewPrice   float64 // m_newprice
	OldPrice   float64 // m_oldprice
	TypeOfOrder TypeOfOrder  // m_typeOfOrder
	OrdType    OrderHitType  // m_ordType
	Status     OrderStatus   // m_status
	Side       TransactionType // m_side
}

// NewOrderStats 创建初始化的 OrderStats
// C++: ExtraStrategy.cpp:238-263 中 new OrderStats() 后的初始化
func NewOrderStats(orderID uint32, side TransactionType, price float64, qty int32,
	typeOfOrder TypeOfOrder, ordType OrderHitType) *OrderStats {
	return &OrderStats{
		Active:      false,
		New:         true,
		Cancel:      false,
		ModifyWait:  false,
		Modify:      0,
		Status:      StatusNewOrder,
		Price:       price,
		Side:        side,
		OrderID:     orderID,
		Qty:         qty,
		OpenQty:     qty,
		DoneQty:     0,
		QuantBehind: 0,
		TypeOfOrder: typeOfOrder,
		OrdType:     ordType,
	}
}

File: tbsrc-java/src/main/java/com/quantlink/trader/core/OrderStats.java
package com.quantlink.trader.core;

import com.quantlink.trader.shm.Constants;

/**
 * 订单生命周期状态追踪。
 * 迁移自: tbsrc/Strategies/include/ExecutionStrategyStructs.h
 *
 * C++ OrderStats struct + OrderStatus enum + OrderHitType enum
 */
public class OrderStats {

    // ---- 枚举: 订单状态 ----
    // 迁移自: ExecutionStrategyStructs.h — enum OrderStatus
    public enum Status {
        NEW_ORDER,
        NEW_CONFIRM,
        NEW_REJECT,
        MODIFY_ORDER,
        MODIFY_CONFIRM,
        MODIFY_REJECT,
        CANCEL_ORDER,
        CANCEL_CONFIRM,
        CANCEL_REJECT,
        TRADED,
        INIT
    }

    // ---- 枚举: 订单命中类型 ----
    // 迁移自: ExecutionStrategyStructs.h — enum OrderHitType
    public enum HitType {
        STANDARD,
        IMPROVE,
        CROSS,
        DETECT,
        MATCH
    }

    // ---- 字段 ----
    // 迁移自: ExecutionStrategyStructs.h — struct OrderStats 全部字段

    public boolean active;
    public boolean isNew;          // C++: m_new
    public boolean modifyWait;     // C++: m_modifywait
    public boolean cancel;         // C++: m_cancel
    public int modifyCount;        // C++: m_modify
    public long lastTS;            // C++: m_lastTS (uint64_t)
    public int orderID;            // C++: m_orderID (uint32_t → Java int)
    public int oldQty;             // C++: m_oldQty
    public int newQty;             // C++: m_newQty
    public int qty;                // C++: m_Qty
    public int openQty;            // C++: m_openQty
    public int cxlQty;             // C++: m_cxlQty
    public int doneQty;            // C++: m_doneQty
    public double quantAhead;      // C++: m_quantAhead
    public double quantBehind;     // C++: m_quantBehind
    public double price;           // C++: m_price
    public double newPrice;        // C++: m_newprice
    public double oldPrice;        // C++: m_oldprice
    public int typeOfOrder;        // C++: m_typeOfOrder (TypeOfOrder enum) — 使用 Constants 值
    public HitType hitType;        // C++: m_ordType (OrderHitType)
    public Status status;          // C++: m_status (OrderStatus)
    public byte side;              // C++: m_side (TransactionType) — Constants.SIDE_BUY/SELL

    public OrderStats() {
        this.status = Status.INIT;
        this.hitType = HitType.STANDARD;
        this.side = Constants.SIDE_BUY;
    }

    @Override
    public String toString() {
        return String.format("Order[id=%d, px=%.2f, qty=%d, open=%d, done=%d, status=%s, hit=%s, side=%s]",
                orderID, price, qty, openQty, doneQty, status,
                hitType, side == Constants.SIDE_BUY ? "BUY" : "SELL");
    }
}
